lib/ecto/multi.ex

```diff
@@ -99,6 +99,7 @@ defmodule Ecto.Multi do
   defstruct operations: [], names: MapSet.new
 
   @type run :: (t -> {:ok | :error, any}) | {module, atom, [any]}
+  @type run_changeset :: (Map.t -> Changeset.t | Ecto.Schema.t)
```

Comment:
the problem with `changeset_fun` is that it returns either `Changeset` or `Schema`. Perhaps calling it `insert_fun/delete_fun` would be better? Naming is hard :)

Also, I think `changeset_fun` type shouldn't be used for the `Ecto.Multi.update` as updates support only changeset and not structs. See: https://github.com/elixir-lang/ecto/blob/v2.0.0-rc.3/lib/ecto/repo/schema.ex#L203